src/processor/InstanceProcessor.ts

```diff
@@ -24,9 +26,48 @@ export class InstanceProcessor {
     }
   }
 
-  static process(input: any, implementationGuide: any): ExportableInstance {
+  static extractRules(input: any, target: ExportableInstance, fisher: utils.Fishable): void {
+    const newRules: ExportableInstance['rules'] = [];
+    // Clone input so it can be modified
+    const inputJSON = cloneDeep(input);
+    const instanceOfJSON = fisher.fishForFHIR(
+      target.instanceOf,
+      utils.Type.Resource,
+      utils.Type.Profile,
+      utils.Type.Extension,
+      utils.Type.Type
+    );
+
+    if (instanceOfJSON == null) {
+      logger.error(
+        `InstanceOf definition not found for ${input.id} and cannot export any Assignment Rules.`
+      );
+      return;
+    }
```

Comment:
I think that emitting an error here is correct, but it might be fine to proceed and add the assignment rules anyway. The value of having `instanceOfJSON` is it lets you know what values on the instance don't need rules. If `instanceOfJSON` isn't found, then all of the non-ignored property values on the instance will get rules.

Reply:
I thought about this and was a little unsure of the way to go on this, but I decided to not process the rules because we also use the `instanceOfJSON` to determine the types of the elements when adding the assignment rules. So without the `instanceOfJSON` we won't know if a string value in the FHIR JSON should really be a FSHCode or if it should stay a string. GoFSH could potentially make those rules, but then if we ran SUSHI on that FSH, SUSHI will log errors that it cannot fix values to mismatched types. I thought it was better to not add assignment rules that could be incorrect rather and avoid those SUSHI errors, but if you think it's better to add the rules regardless, I can make those changes.

Reply:
I keep going back and forth on "don't write things when we can't apply the proper logic" vs "write as much as you can, and emit errors when we can't apply the proper logic." I think in this case, since it's specifically _not_ a FHIR validation issue, I'm back on your side of not writing any rules. The error is not FHIR validity, it's GoFSH validity.

Reply:
Hmmm... I had the same thought at first (that you can't output an instance at all if you can't find the `InstanceOf` -- because you need to know the types for the paths).  BUT... I think we actually _can_ find the types for the paths.  _Every_ instance should have a `resourceType` field, and the `resourceType` should _always_ be a valid FHIR resource.  Profiles cannot _add_ new types or _change_ existing types; they can only _narrow_ types.  Therefore, no matter what the profile is, we should be able to resolve the types for every element by using the resource from the `resourceType`.

So, it seems to me that if something is an instance of a profile, we should try to find the profile first.  But if that fails, we should fall back to the `resourceType` -- which we should _never_ fail.  If we fall back to the `resourceType`, however, we should issue a warning that we couldn't find the profile, so the generated instance may not be the optimal representation.

Reply:
Ah that's a good point! I've implemented that in 7528e9b